### PR TITLE
Function to convert manifest.json to renv.lock

### PR DESCRIPTION
**Pull request by JosiahParry**
Overview: 

- Many customers are trying to adopt {renv} to improve their development process. Customers will deploy a piece of content to RSC using push button deployment. They'll come back to the project and learn that their environment is wrecked. The RSC content bundle contains the manifest.json file generated from rsconnect. Converting that to an renv.lock file is not straight forward. This pull request creates functionality to take the manifest.json file and convert it to

## Changes: 

- Adds a `version()` method to the internal `lockfile` R6 object
- Creates new function `rsc_manifest_to_renv_lock()` which takes a manifest as input and produces an renv.lockfile.

### CODE REVIEW

File: R/manifest-convert.R [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+#' @details
+#'
+#' By default the `lockfile` argument is set to `NA`. This will not create a new renv.lock file.
+#' Rather, it will return a lockfile object (see `?lockfile`) that can be used to create a new
+#' renv.lock file. If `lockfile` is set to a character string, a new file will be created with that
+#' path—e.g. `renv.lock`—and the lockfile object will be returned.
+#'
+#'
+#' @export
+rsc_manifest_to_renv_lock <- function(manifest = NULL, lockfile = NA) {
```
Comment (kevinushey) [inline R/manifest-convert.R:25]: Could we make this an un-exported function called e.g. `renv_lockfile_from_manifest()`?

File: R/manifest-convert.R [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  }
+
+  # create lockfile if non-existant and lockfile != NA
+  if (!file.exists(lockfile %&&% "") & !is.na(lockfile)) {
+    fmt <- "%s doesn't exist. Creating new file."
+    vwritef(fmt, lockfile)
+    file.create(lockfile)
+  }
+
+  cat("* Reading manifest.json...\n")
```
Comment (kevinushey) [inline R/manifest-convert.R:38]: Use `vwritef()` (if you think the user should be able to disable logging) or `writef()` instead.

File: R/manifest-convert.R [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+#'
+#' @export
+rsc_manifest_to_renv_lock <- function(manifest = NULL, lockfile = NA) {
+
+  if (missing(manifest)) {
+    stop("Provide the path the manifest.json file.", call. = FALSE)
+  }
+
+  # create lockfile if non-existant and lockfile != NA
+  if (!file.exists(lockfile %&&% "") & !is.na(lockfile)) {
```
Comment (kevinushey) [inline R/manifest-convert.R:32]:
```suggestion
  if (!is.na(lockfile) && file.exists(lockfile)) {
  if (!file.exists(lockfile %&&% "") & !is.na(lockfile)) {
```

File: R/manifest-convert.R [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    vwritef(fmt, lockfile)
+    file.create(lockfile)
+  }
+
+  cat("* Reading manifest.json...\n")
+  .manifest <- renv_json_read(manifest)
+
+  lock <- lockfile(lockfile)
+
+  cat("* Parsing dependencies...\n")
```
Comment (kevinushey) [inline R/manifest-convert.R:43]: Same as above.

File: R/manifest-convert.R [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  manifest_repos <- as.list(unlist(unique(unname(all_pkg_sources))))
+
+  # make updates
+  do.call(lock$repos, manifest_repos)
+  do.call(lock$add, manifest_pkgs)
+  lock$version(.manifest[["platform"]])
+
+  if (is.na(lockfile)) {
+    return(lock)
+  } else{
```
Comment (kevinushey) [inline R/manifest-convert.R:59]:
```suggestion
  } else {
```

File: NAMESPACE [hunk 6]
```diff
@@ -28,6 +28,7 @@ export(rehash)
 export(remove)
 export(restore)
 export(revert)
+export(rsc_manifest_to_renv_lock)
```
Comment (kevinushey) [inline NAMESPACE:31]: I'd prefer to leave this un-exported.

**Review by kevinushey — CHANGES_REQUESTED** [review]
Thanks for the PR! I'd prefer to leave this as an unexported function for now (we could document and allow users to call it with `renv:::` explicitly), since this function is fairly specific + coupled to RSC.

**Comment by JosiahParry** [comment]
Thanks, Kevin! The requested changes have been made.

**Comment by kevinushey** [comment]
Thanks!

**Comment by JosiahParry** [comment]
No, thank you! :)

**Pull request merged**